packages/rest/src/providers/log-error-provider.ts

```diff
@@ -6,18 +6,20 @@
 import {Provider} from '@loopback/context';
 import {ServerRequest} from 'http';
 import {LogError} from '../internal-types';
+import * as debugModule from 'debug';
+const debug = debugModule('loopback:rest:error');
 
 export class LogErrorProvider implements Provider<LogError> {
   value(): LogError {
     return (err, statusCode, req) => this.action(err, statusCode, req);
   }
 
   action(err: Error, statusCode: number, req: ServerRequest) {
-    if (statusCode < 500) {
+    if (statusCode < 400) {
       return;
     }
 
-    console.error(
+    debug(
```

Comment:
-1

Printing failed HTTP requests to `stderr` is intentional behavior that needs to be preserved as the default mode of LoopBack4+ applications.

This is important for our unit-tests too, because when a request fails with an HTTP status code we did not expect, then we do want to see full stack trace of the server error in the test output. For example, in a test checking that a request for unknown path returns 404, we want to see server stack trace when the request failed with 500 Internal Error caused e.g. by `TypeError` or `ReferenceError`.

We already have a solution for suppressing these messages you can use for inspiration, see here:

https://github.com/strongloop/loopback-next/blob/ff4a320825720270346013ce81174100bef28f37/packages/rest/test/integration/http-handler.integration.ts#L503-L517